### ISSUE TITLE
fs.readFileSync error for .md markdown files

### ISSUE THREAD

**Issue by quinton-ashley** [jprichardson/node-fs-extra#593]
- **Operating System:macOS 10.14 Developer Beta**
- **Node.js version:8.11.1**
- **`fs-extra` version: 4.0.3**

Upon reading a markdown file with fs.readFileSync() I get an JSON parsing error.  This markdown file does not contain JSON.
```
Uncaught SyntaxError: /Users/qashto/Documents/dev/lib/qodemate-test-files/jsTestFolder/lesson.md: Unexpected token # in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.readFileSync (/Users/qashto/Documents/dev/apps/qodemate/node_modules/jsonfile/index.js:63)
    at module.exports.open (/Users/qashto/Documents/dev/apps/qodemate-core/index.js:84)
    at chooseFile (/Users/qashto/Documents/dev/apps/qodemate-desktop/views/js/renderer.js:49)
    at HTMLButtonElement.bigButtonClicked (/Users/qashto/Documents/dev/apps/qodemate-desktop/views/js/renderer.js:58)
    at HTMLButtonElement.dispatch (/Users/qashto/Documents/dev/apps/qodemate/node_modules/jquery/dist/jquery.js:5183)
    at HTMLButtonElement.elemData.handle (/Users/qashto/Documents/dev/apps/qodemate/node_modules/jquery/dist/jquery.js:4991)
```

**Comment by JPeer264** [comment]
Hm, I just tried following code and it worked:

```js
const fs = require('fs-extra')
const myFile = fs.readFileSync('./helloworld.md', 'utf8')

console.log(myFile) // content of helloworld.md
```

Are you sure that you not accidentally wrote `fs.readJsonSync`?

**Comment by quinton-ashley** [comment]
jsonfile has a readFileSync, is it possible it's getting overriden?
```javascript
const fs = require('fs-extra');
let data = fs.readFileSync('/Users/qashto/Documents/dev/lib/qodemate-test-files/jsTestFolder/lesson.md', 'utf8');
```

**Comment by JPeer264** [comment]
Actually not a bad theory, but we ensured that we exported just the necessary functions (as of [4.0.3](https://github.com/jprichardson/node-fs-extra/blob/bd3376b78fb880a7539825d6c320458221977c91/lib/json/jsonfile.js)).

I am really sorry, but I can't reproduce this issue. Can you maybe post the output of the following code (just to be sure):

```js
console.log(fs.readFileSync.toString())
```

**Comment by quinton-ashley** [comment]
Nevermind, this was not an issue with fs-extra.  I'm working on a package called [sidecar](https://github.com/quinton-ashley/sidecar) that overrides require to allow packages to be required outside node_modules, "sidecarred", during development. I made an error that allowed './fs' in the index.js of fs-extra to refer to the native fs module I think.   I fixed it by not allowing strings to start with './' for sidecarred packages.  Fixed in [1.0.7](https://github.com/quinton-ashley/sidecar/commit/f61be46d3e0e7d389849a98152012fa73badfe85#diff-168726dbe96b3ce427e7fedce31bb0bc)
